nengo_gui/static/components/spa_similarity.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * Line graph showing semantic pointer decoded values over time
+ * @constructor
+ *
+ * @param {dict} args - A set of constructor arguments (see Nengo.Component)
+ * @param {int} args.n_lines - number of decoded values
+ * @param {float} args.min_value - minimum value on y-axis
+ * @param {float} args.max_value - maximum value on y-axis
+ * @param {Nengo.SimControl} args.sim - the simulation controller
+ */
+
+Nengo.SpaSimilarity = function(parent, sim, args) {
+    Nengo.Value.call(this, parent, sim, args);
+
+    this.data_store = new Nengo.VariableDataStore(this.n_lines, this.sim, args.synapse);
+    this.show_pairs = false;
+
+    var self = this;
+
+    this.colors = Nengo.make_colors(6);
+    this.color_func = function(d, i){return self.colors[i%6]};
+
+    // create the legend from label args
+    this.pointer_labels = args.pointer_labels;
+    this.legend = document.createElement('div');
+    this.legend.classList.add('legend', 'unselectable');
+    this.div.appendChild(this.legend);
+    this.legend_svg = Nengo.draw_legend(this.legend, args.pointer_labels, this.color_func);
+};
+
+Nengo.SpaSimilarity.prototype = Object.create(Nengo.Value.prototype);
+Nengo.SpaSimilarity.prototype.constructor = Nengo.SpaSimilarity;
+
+Nengo.SpaSimilarity.prototype.show_pairs_toggle = function(new_labels){
+    // clear the database and make a new one
+    self.data_store = new Nengo.VariableDataStore()
+
+    // delete the legend's children
+    while(this.legend.lastChild){
+        this.legend.removeChild(this.legend.lastChild);
+    }
+
+    // redraw all the legends
+    this.pointer_labels = new_labels;
+    this.legend_svg = Nengo.draw_legend(this.legend, new_labels, this.color_func);
+
+}
+
+Nengo.SpaSimilarity.prototype.data_msg = function(push_data){
+
+    var data_dims = push_data.length - 1;
+
+    if(data_dims !== this.n_lines){
+      this.data_store.dims = data_dims;
+      this.n_lines = data_dims;
+    }
+
+    this.data_store.push(push_data);
+    this.schedule_update();
+};
+
+Nengo.SpaSimilarity.prototype.update_legend = function(new_labels){
+    // Should figure out how to mix recs and text into one
+
+    var self = this;
+    this.pointer_labels = this.pointer_labels.concat(new_labels);
+
+    // expand the svg
+    this.legend_svg.attr("height", 20*this.pointer_labels.length);
```

Comment:
It also appears below, so it should probably be at least a variable to be able to adjust it in one place.

Also unsure about the other magic numbers in there. I haven't tested it yet, but I'm worried that it might break for different font sizes.